subprojects/ivy/src/testFixtures/groovy/org/gradle/api/publish/ivy/AbstractIvyPublishIntegTest.groovy

```diff
@@ -17,37 +17,97 @@
 package org.gradle.api.publish.ivy
 
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.ExperimentalFeaturesFixture
+import org.gradle.test.fixtures.GradleMetadataAwarePublishingSpec
 import org.gradle.test.fixtures.ivy.IvyFileModule
+import org.gradle.test.fixtures.ivy.IvyJavaModule
+import org.gradle.test.fixtures.ivy.IvyModule
 
 import static org.gradle.integtests.fixtures.RepoScriptBlockUtil.mavenCentralRepositoryDefinition
 
-class AbstractIvyPublishIntegTest extends AbstractIntegrationSpec {
+class AbstractIvyPublishIntegTest extends AbstractIntegrationSpec implements GradleMetadataAwarePublishingSpec {
 
-    protected def resolveArtifacts(IvyFileModule module) {
+    def setup() {
+        prepare()
+    }
+
+    ResolveParams resolveParams(Map args = [:]) {
+        new ResolveParams(args)
+    }
+
+    protected static IvyJavaModule javaLibrary(IvyFileModule module) {
+        new IvyJavaModule(module)
+    }
+
+    protected def resolveArtifacts(IvyModule module) {
         doResolveArtifacts("group: '${sq(module.organisation)}', name: '${sq(module.module)}', version: '${sq(module.revision)}'")
     }
 
-    protected def resolveArtifacts(IvyFileModule module, def configuration) {
+    protected def resolveArtifacts(IvyModule module, String configuration) {
         doResolveArtifacts("group: '${sq(module.organisation)}', name: '${sq(module.module)}', version: '${sq(module.revision)}', configuration: '${sq(configuration)}'")
     }
 
-    protected def resolveArtifactsWithStatus(IvyFileModule module, def status) {
-        doResolveArtifacts("group: '${sq(module.organisation)}', name: '${sq(module.module)}', version: '${sq(module.revision)}'", status)
+    protected def resolveAdditionalArtifacts(IvyJavaModule module) {
+        doResolveArtifacts("group: '${sq(module.organisation)}', name: '${sq(module.module)}', version: '${sq(module.revision)}'", resolveParams(additionalArtifacts: module.additionalArtifacts))
     }
 
-    private def doResolveArtifacts(def dependency, def status=null) {
+    protected def resolveArtifactsWithStatus(IvyModule module, String status) {
+        doResolveArtifacts("group: '${sq(module.organisation)}', name: '${sq(module.module)}', version: '${sq(module.revision)}'", resolveParams(status:status))
+    }
+
+    private def doResolveArtifacts(String dependency, ResolveParams params = resolveParams()) {
         // Replace the existing buildfile with one for resolving the published module
         settingsFile.text = "rootProject.name = 'resolve'"
+
+        if (resolveModuleMetadata) {
+            ExperimentalFeaturesFixture.enable(settingsFile)
+        } else {
+            executer.beforeExecute {
+                // Remove the experimental flag set earlier...
+                // TODO:DAZ Remove this once we support excludes and we can have a single flag to enable publish/resolve
```

Comment:
I think this is not needed anymore? I can not find any Ivy test doing `disableModuleMetadataPublishing()`. Or did I misunderstand the comment?

Reply:
There was something odd going on. I'll investigate.

Reply:
No you're right, it shouldn't be necessary anymore, now that excludes are implemented.